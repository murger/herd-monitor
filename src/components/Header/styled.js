import styled from 'styled-components';

export const Masthead = styled.header`
  width: 100%;
  padding: 32px;
  box-sizing: border-box;
  display: flex;
  align-items: flex-end;
  justify-content: space-between;
`;

export const Logo = styled.img`
  height: 48px;
`;

export const Typeface = styled.div`
  display: inline-block;
  margin-left: 8px;
  color: #090D45;
  font-size: 24px;
  font-weight: 900;
  line-height: 0.9;
  letter-spacing: ${-1};
  vertical-align: baseline;
`;
