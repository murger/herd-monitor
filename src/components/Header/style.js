import styled from 'styled-components';

export const Masthead = styled.header`
  width: 100%;
  padding: 32px;
  padding-top: 48px;
  box-sizing: border-box;
  display: flex;
  align-items: flex-end;
  justify-content: space-between;
`;

export const Logo = styled.img`
  height: 48px;
`;

export const Typeface = styled.div`
  display: inline-block;
  margin-left: 8px;
  color: ${props => props.theme.navy};
  font-size: 24px;
  font-weight: 900;
  line-height: 0.9;
  letter-spacing: -1px;
  vertical-align: baseline;
`;
