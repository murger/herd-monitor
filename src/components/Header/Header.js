import React from 'react';
import { ServiceConsumer } from '../../contexts/ServiceContext';
import Dropdown from '../Dropdown';
import { Masthead, Logo, Typeface } from './style';
import larry from '../../../pub/larry.svg';

const Header = () => (
  <Masthead>
    <span>
      <Logo src={larry} />
      <Typeface>
				Herd<br />
				Monitor
      </Typeface>
    </span>
    <span>
      <ServiceConsumer>
        {({ sortTransactions }) => (
          <Dropdown
            defaultValue="none"
            onChange={({ target: { value } }) => sortTransactions(value)}>
            <option value="none" disabled>Sort by</option>
            <option value="fromDate">Date</option>
            <option value="status">Status</option>
          </Dropdown>
        )}
      </ServiceConsumer>
    </span>
  </Masthead>
);

export default Header;
