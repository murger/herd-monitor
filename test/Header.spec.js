import React from 'react';
import renderer from 'react-test-renderer';
import Header from '../src/components/Header';

it('renders the correctly', () => {
  const tree = renderer.create(<Header />).toJSON();
  expect(tree).toMatchSnapshot();
});
