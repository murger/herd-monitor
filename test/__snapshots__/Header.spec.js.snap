// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the correctly 1`] = `
<header
  className="sc-htpNat hfZxvQ"
>
  <span>
    <img
      className="sc-bxivhb jSVldU"
      src=""
    />
    <div
      className="sc-ifAKCX fGpzrZ"
    >
      Herd
      <br />
      Monitor
    </div>
  </span>
  <span>
    <div
      className="sc-bdVaJa fRmVhe"
    >
      <select
        className="sc-bwzfXH ejOUFP"
      >
        <option>
          Date
        </option>
        <option>
          Status
        </option>
      </select>
    </div>
  </span>
</header>
`;
